### ISSUE TITLE
Request: Folding Wings

### ISSUE THREAD

**Issue by fsendventd** [DonBruce64/MinecraftTransportSimulator#16]
Many naval aircraft have folding wings to save space in the cramped hangars of aircraft carriers. Once we get moving hitboxes, maybe add a "wing" rotation variable.

**Comment by Nianor** [comment]
Hmm. Could we also get variable-sweep wings with this? Or would that be asking for too much?

**Comment by fsendventd** [comment]
Wouldn't that require changing the drag calculation when you swept/unswept the wings (i.e. lowering wingspan while maintaining same wing area)?

**Comment by fsendventd** [comment]
But for wing stowage, you could just set the span and area to 0, since you have pretty much no wing able to provide lift when they're folded. An F-14, on the other hand, loses half its wingspan but keeps the same area when it goes from minimum sweep to full sweep.

**Comment by Nianor** [comment]
...But you're still changing those values, even if you're setting them to zero. So wouldn't it make sense to have them as part of the same system? Unless there's a variable pitch aircraft that also has folding wings.

**Comment by DonBruce64** [comment]
I'm going to cancel this.  Not because I don't agree with the concept, but because the execution will be flawed to the point that things will be wonky.  This comes down to these points:

1) When wings are folded on aircraft, they are done in such a way that they would become damaged should the aircraft take off.  This means that planes shouldn't be able to fly with them folded, which means that I'd have to find some way to inhibit flight.  But that's nearly impossible to do in MC with lots of hills to push planes off of.

2) If I *don't* inhibit flight, I would need to find a way to adjust the wing area on-the-fly.  That's more JSON work and more model changes for a system that won't be used on most models.

3) Adding this feature would mean another keybind, and we already have gobs of them.

4) Carriers and storage space is a moot point in a game where carriers don't exist, and storage can be made sub-space by putting planes in chests.

TL:DR - Too complex for what it adds and it doesn't fit in MC that well.
